src/app/modules/shared-components/components/message/message.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, OnChanges, Input, SimpleChanges } from '@angular/core';
+import { Message } from 'primeng/api';
+
+@Component({
+  selector: 'alg-message',
+  templateUrl: './message.component.html',
+  styleUrls: ['./message.component.scss']
+})
+export class MessageComponent implements OnChanges {
+
+  @Input() type: 'success' | 'info' | 'error';
+  @Input() summary: string;
+  @Input() detail: string;
+  @Input() closable = true;
```

Comment:
For all your input, you have to consider they might not be given by the component using this one. 
So you define that your type, summary and details and non optional, but in practice, if one of the input is not given, the value will be "undefined", leading to bugs. So you have to give a default value to them the same way you do with "closable" or "msgs".

Reply:
Actually, for summary, that might be what you want, but so in this case, you define it as optional, so `@Input() summary?: string;`